generator/cli/generate-migration-sql.ts

```diff
@@ -0,0 +1,106 @@
+
+export default async function generateMigrationSql(params: {
+  templatesRepoDir: string
+  outputDir: string
+}): Promise<void> {
+  log(`generating migration sql for each template x provider to ${params.outputDir}`)
+
+  FS.remove(params.outputDir)
+
+  const templateInfos = getTemplateInfos({ templatesRepoDir: params.templatesRepoDir })
+
+  log(`Found templates:`, { templates: templateInfos.map((t) => t.displayName) })
+
+  const indexFile = `${params.outputDir}/index.ts`
+  FS.write(indexFile, '')
+
+  const exportsList: string[] = []
+  const providers: DatasourceProvider[] = ['postgresql', 'mysql', 'sqlserver']
+  await Promise.all(
+    providers.map(async (provider) => {
+      await Promise.all(
+        templateInfos.map(async (t) => {
+          const templateName = t.displayName.trim()
+          const schemaPath = `./templates-repo/${templateName}/prisma/schema.prisma`
+          const newSchemaPath = `/tmp/${t.handles.pascal.value}/${provider}/schema.prisma`
+          const filename = `${t.handles.pascal.value}-${provider}.ts`
+          const filenameExtentionless = `${t.handles.pascal.value}-${provider}`
+          const outputPath = params.outputDir + `/${filename}`
+          const exportName = t.handles.pascal.value + provider
+          exportsList.push(exportName)
+          // Replace schema provider for diff command
+          const content = FS.read(schemaPath)
+          if (!content) throw new Error('Could not copy')
+          const modifiedSchema = replaceProvider(provider, { content: content, path: schemaPath })
+          // Write modified schema to disk
+          FS.remove(newSchemaPath)
+          FS.file(newSchemaPath)
+          FS.write(newSchemaPath, modifiedSchema)
+          log(`Wrote modified template schema to ${newSchemaPath}`)
+          // Make sure output path exists
+          FS.file(outputPath)
+          const res = execa.commandSync(
+            `yarn prisma migrate diff --preview-feature --from-empty --to-schema-datamodel ${newSchemaPath}  --script`,
+            {
+              cwd: process.cwd(),
+              env: {
+                DATABASE_URL: connectionString(provider),
+              },
+            }
+          )
+          // Remove command that is returned in migrate diff response
+          const substr = '--script'
+          const commandIndexEnd = res.stdout.indexOf(substr)
+          const formatted = splitSql(res.stdout.substr(commandIndexEnd + substr.length))
+          // Write sql to disk as a ts object default export
+          FS.write(outputPath, formatted)
+          await prependFile(outputPath, `const ${exportName} = `)
+          FS.append(outputPath, `;export default ${exportName}`)
+          // Add individual exports to generatedMigrations/index.ts
+          FS.append(indexFile, `import ${exportName} from "./${filenameExtentionless}"; \n`)
+          log(`Output to ${outputPath}`)
```

Comment:
Manipulate the source code in memory until you've got your desired result, then flush it to disk once.